File: 3/R/funcs.R
get_primes <- function(limit) {
    sieve = rep(TRUE, limit)
    primes = c()

    sieve[1] = FALSE

    for (k in 1:length(sieve)) {
        if(sieve[k]) {
            primes = append(primes, k)
            sieve[seq(from = k, by = k, to = limit)] = FALSE
        }
    }
    return(as.numeric(primes))
}


get_largest_factor <- function(n) {
    biggest_prime = 2
    primes = get_primes(ceiling(sqrt(n)))
        for (prime in primes) {
            while (n %% prime == 0) {
                n = n / prime
                biggestprime = prime
            }
        }
    if (n == 1) return(biggestprime)
    return(n)
}

File: 3/JS/funcs.js
function find_primes(limit) {
    var sieve = [];
    var primes = [];

    for (let i = 0; i <= limit; i++) {
        sieve.push(true);
    }
    sieve[0] = sieve[1] = false;

    for (let [i, isprime] of sieve.entries()) {
        if(isprime) {
            primes.push(i);
            if(i <= Math.ceil(Math.sqrt(limit))) {
                for(let k = i * i; k <= limit; k += i) {
                    sieve[k] = false;
                };
            };
        }
    }
    return primes;
};


function find_biggest_factor(n) {
    var primes = find_primes(Math.ceil(Math.sqrt(n)));
    var biggestfactor = null;
    for(let [i, prime] of primes.entries()) {
        if(n % prime === 0) {
            biggestfactor = prime;
        }
    }
    if(biggestfactor === null){
        return n;
    }
    return biggestfactor;
}
